subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -1104,24 +1143,47 @@ uint32_t radio_ccm_mic_is_valid(void)
 
 static uint8_t MALIGN(4) _aar_scratch[3];
 
-void radio_ar_configure(uint32_t nirk, void *irk)
+void radio_ar_configure(uint32_t nirk, void *irk, uint8_t phy)
 {
+	uint32_t addrptr;
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT) &&
+	    IS_ENABLED(CONFIG_BT_CTLR_PHY_CODED) &&
+	    (phy == BIT(2))) {
+		addrptr = NRF_RADIO->PACKETPTR + 4;
+	} else {
+		addrptr = NRF_RADIO->PACKETPTR - 1;
+	}
+
 	NRF_AAR->ENABLE = (AAR_ENABLE_ENABLE_Enabled << AAR_ENABLE_ENABLE_Pos) &
 			  AAR_ENABLE_ENABLE_Msk;
 	NRF_AAR->NIRK = nirk;
 	NRF_AAR->IRKPTR = (uint32_t)irk;
-	NRF_AAR->ADDRPTR = (uint32_t)NRF_RADIO->PACKETPTR - 1;
+	NRF_AAR->ADDRPTR = addrptr;
 	NRF_AAR->SCRATCHPTR = (uint32_t)&_aar_scratch[0];
 
 	NRF_AAR->EVENTS_END = 0;
 	NRF_AAR->EVENTS_RESOLVED = 0;
 	NRF_AAR->EVENTS_NOTRESOLVED = 0;
 
-	radio_bc_configure(64);
+	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT)) {
+		if (IS_ENABLED(CONFIG_BT_CTLR_PHY_CODED) && (phy == BIT(2))) {
+			radio_bc_configure(48);
+		} else {
+			radio_bc_configure(8);
+		}
+	} else {
+		radio_bc_configure(64);
+	}
 	radio_bc_status_reset();
 
 	hal_trigger_aar_ppi_config();
-	hal_radio_nrf_ppi_channels_enable(BIT(HAL_TRIGGER_AAR_PPI));
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_ADV_EXT)) {
+		nrf_radio_int_enable(NRF_RADIO, RADIO_INTENSET_BCMATCH_Msk);
```

Comment:
Do not use BCMATCH ISR, for many reasons, more CPU use per PDU reception, dependency on CPU speed and OS ISR latencies to setup bit count compare in time, and higher peak current due to simultaneous radio-CPU use.

Thinking aloud, options, use the single NRF_AAR for either legacy PDU bit count compare or for common extended advertising payload format bit count compare. Have this as Kconfig. On Disabled ISR, use NRF_AAR again inline if the PPI based was incorrect. i.e. if PPI based was for legacy PDU and received PDU was extended, then inside the ISR resolve the address using NRF_AAR without IRQ (spin wait with low power, WFE).

Reply:
Expected PDU type can be implied from channel (pri/sec) and PHY. The only case where we can receive both types is 1M on primary so we need to use something by default. This could be configurable as Kconfig, although since I do not really see how one can decide which is better, for now I left this "hardcoded" to legacy and then resolve inline for extended. Adding a Kconfig should be pretty simple if needed.